tests/std/tests/P2321R2_views_adjacent/test.cpp

```diff
@@ -899,6 +898,62 @@ constexpr void instantiation_test() {
 #endif // TEST_EVERYTHING
 }
 
+// LWG-4098 views::adjacent<0> should reject non-forward ranges
+template <size_t N, ranges::input_range Rng>
+constexpr void test_input_only(Rng&&) {
+    if constexpr (!ranges::forward_range<Rng>) {
+        static_assert(!CanViewAdjacent<Rng&, N>);
+        static_assert(!CanViewAdjacent<Rng, N>);
+        static_assert(!CanConstructAdjacentView<Rng&, N>);
+        static_assert(!CanConstructAdjacentView<Rng, N>);
+    }
+
+    if constexpr (!ranges::forward_range<const Rng>) {
+        static_assert(!CanViewAdjacent<const Rng&, N>);
+        static_assert(!CanViewAdjacent<const Rng, N>);
+        static_assert(!CanConstructAdjacentView<const Rng&, N>);
+        static_assert(!CanConstructAdjacentView<const Rng, N>);
+    }
+}
+
+struct input_only_instantiator {
+#ifdef TEST_EVERYTHING
+    template <ranges::input_range R>
+    static constexpr void call() {
+        R r{some_ints};
+        test_input_only<0>(r);
+        test_input_only<1>(r);
+        test_input_only<2>(r);
+        test_input_only<4>(r);
+        test_input_only<7>(r);
+    }
+#else // ^^^ test all input range permutations / test only "interesting" permutations vvv
+    template <class Tag, test::Common IsCommon, test::Sized IsSized>
+    static constexpr void call() {
+        test_range<const int, Tag, IsCommon, IsSized> r{some_ints};
+        test_input_only<0>(r);
+        test_input_only<1>(r);
+        test_input_only<2>(r);
+        test_input_only<4>(r);
+        test_input_only<7>(r);
+    }
+#endif // TEST_EVERYTHING
+};
+
+constexpr void instantiation_input_only_test() {
+#ifdef TEST_EVERYTHING
+    test_in<input_only_instantiator, const int>();
+#else // ^^^ test all forward range permutations / test only "interesting" permutations vvv

```

Comment:
Unless I'm very confused, this was copy-pasted and should instead say "test all input range permutations".